src/Rules/Drupal/Tests/BrowserTestBaseDefaultThemeRule.php

```diff
@@ -30,6 +30,27 @@ public function processNode(Node $node, Scope $scope): array
             return [];
         }
 
+        // Only inspect tests.
+        // @todo replace this str_ends_with() when php 8 is required.
+        if (0 === substr_compare($node->namespacedName->getLast(), 'Test', -4)) {
+            return [];
+        }
+
+        // Do some cheap preflight tests to make sure the class is in a
+        // namespace that makes sense to inspect.
+        $parts = $node->namespacedName->parts;
+        // The namespace is too short to be a test so skip inspection.
+        if (count($parts) < 3) {
+            return [];
+        }
+        // If the 4th component matches it's a module test. If the 2nd, core.
+        if ($parts[3] !== 'Functional' &&
+            $parts [3] !== 'FunctionalJavascript' &&
+            $parts[1] !== 'FunctionalTests' &&
+            $parts[1] !== 'FunctionalJavascriptTests') {
+            return [];
```

Comment:
These need to be split up, I think. Because it's `&&`. So should it actually be all `||`? So that the first `true` exits the conditional?

Also, I've taken the style from PHPStan of putting `&&` or `||` first, so:

```php
        if ($parts[3] !== 'Functional'
            || $parts [3] !== 'FunctionalJavascript'
            || $parts[1] !== 'FunctionalTests'
            || $parts[1] !== 'FunctionalJavascriptTests') {
```

Reply:
I think the logic is actually correct but it is just one of those confusing inversions because we're providing a firewall.

We want to continue if one of the namespaces match which is to say `a == 'Functional' || a == 'FunctionalJavascript' || b == 'FunctionalTests' || b == 'FunctionalJavascriptTests'`

and the logic inversion of `a ==  b || c == d` is `a != b && c != d` so we get the && rule.

I'll change the formatting though.